src/dsp_tools/utils/connection_live.py

```diff
@@ -232,7 +235,7 @@ def post(
         if jsondata:
             # if data is not encoded as bytes, issues can occur with non-ASCII characters,
             # where the content-length of the request will turn out to be different from the actual length
-            data = jsondata.encode("utf-8") if jsondata else None
+            data = json.dumps(jsondata, cls=SetEncoder).encode("utf-8")
             request = partial(request, data=data)
```

Comment:
with this PR you practically already did the work for https://www.notion.so/dasch-swiss/New-Improve-JSON-serialization-for-HTTP-requests-c344eea38112401d8ecbcc2a78998983?pvs=4  
if you want to keep that for a separate PR, sure. But I wouldn't let it wait, since the heavy lifting is already done, which makes this really low hanging fruit.